services/ecs.py

```diff
@@ -2,52 +2,226 @@
+
+    for task_definition in latest_task_definitions.keys():
+        task_definition_arn = f"{task_definition}:{latest_task_definitions[task_definition]}"
+        task_definition = client.describe_task_definition(taskDefinition=task_definition_arn)["taskDefinition"]
+        containers = task_definition["containerDefinitions"]
+
+        for container in containers:
+            if container.get("privileged") == True:
+                non_compliant_resources.append(task_definition["taskDefinitionArn"])
+                break
+        else:
+            compliant_resources.append(task_definition["taskDefinitionArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def ecs_containers_readonly_access():
+    compliant_resources = []
+    non_compliant_resources = []
+    task_definitions = client.list_task_definitions(status="ACTIVE")["taskDefinitionArns"]
+    latest_task_definitions = {}
+
+    for task_definition in task_definitions:
+        if latest_task_definitions.get(task_definition.rsplit(":", 1)[0], 0) < int(task_definition.rsplit(":", 1)[1]):
+            latest_task_definitions[task_definition.rsplit(":", 1)[0]] = int(task_definition.rsplit(":", 1)[1])
+
+    for task_definition in latest_task_definitions.keys():
+        task_definition_arn = f"{task_definition}:{latest_task_definitions[task_definition]}"
+        task_definition = client.describe_task_definition(taskDefinition=task_definition_arn)["taskDefinition"]
+        containers = task_definition["containerDefinitions"]
+
+        for container in containers:
+            if container.get("readonlyRootFilesystem") == False:
+                non_compliant_resources.append(task_definition["taskDefinitionArn"])
+                break
+        else:
+            compliant_resources.append(task_definition["taskDefinitionArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def ecs_container_insights_enabled():
+    compliant_resources = []
+    non_compliant_resources = []
+    cluster_arns = client.list_clusters()["clusterArns"]
+
+    for cluster_arn in cluster_arns:
+        clusters = client.describe_clusters(clusters=[cluster_arn], include=["SETTINGS"])["clusters"]
+
+        for cluster in clusters:
+            settings = cluster["settings"]
+
+            for setting in settings:
+                if setting["name"] == "containerInsights" and setting["value"] == "enabled":
+                    compliant_resources.append(cluster_arn)
+                    break
+            else:
+                non_compliant_resources.append(cluster_arn)
```

Comment:
for문이 너무 많아질 경우 가독성이 크게 떨어지게 됩니다. 또한, `describe_clusters` API call 하나만으로 모든 처리가 가능합니다.
```suggestion

    clusters = client.describe_clusters(include=["SETTINGS"])["clusters"]

    for cluster in clusters:
        container_insights_setting = [setting for setting in cluster["settings"] if setting["name"] == "containerInsights"]

        if container_insights_setting and container_insights_setting[0]["value"] == "enabled":
            compliant_resources.append(cluster["clusterArn"])
        else:
            non_compliant_resources.append(cluster["clusterArn"])
```